pkg/reconciler/revision/cruds_test.go

```diff
@@ -0,0 +1,381 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package revision
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	corev1informer "k8s.io/client-go/informers/core/v1"
+	kubefake "k8s.io/client-go/kubernetes/fake"
+	"knative.dev/pkg/ptr"
+	"knative.dev/serving/pkg/apis/serving"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+)
+
+var (
+	containerName = "my-container-name"
+
+	defaultRevision = &v1alpha1.Revision{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "foo",
+			Name:      "bar",
+			UID:       "1234",
+			Labels: map[string]string{
+				serving.ConfigurationLabelKey: "cfg",
+				serving.ServiceLabelKey:       "svc",
+				serving.RouteLabelKey:         "im-a-route",
+			},
+		},
+		Spec: v1alpha1.RevisionSpec{
+			DeprecatedContainer: &corev1.Container{
+				Name:  containerName,
+				Image: "busybox",
+			},
+			RevisionSpec: v1.RevisionSpec{
+				TimeoutSeconds: ptr.Int64(45),
+			},
+		},
+	}
+)
+
+func refInt64(num int64) *int64 {
+	return &num
+}
+
+type containerOption func(*corev1.Container)
+type revisionOption func(*v1alpha1.Revision)
+
+func container(container *corev1.Container, opts ...containerOption) corev1.Container {
+	for _, option := range opts {
+		option(container)
+	}
+	return *container
+}
+
+func withReadinessProbe(handler corev1.Handler) containerOption {
+	return func(container *corev1.Container) {
+		container.ReadinessProbe = &corev1.Probe{Handler: handler}
+	}
+}
+
+func withTCPReadinessProbe() containerOption {
```

Comment:
Why is this relevant for these tests? Seems to me as if the container should just be part of the default revision and your test code should only insert parts relevant for this test?